### PR TITLE
[MODUSERS-355]-Added consumer for delete event.

### PR DESCRIPTION
**Pull request by SinghAdes**
https://issues.folio.org/browse/MODUSERS-355

We need to implement a consumer of the CONSORTIUM_PRIMARY_AFFILIATION_DELETED event that is produced by the mod-consortia module. This event will include userId, tenantId in the payload and record that matches userId + tenantId should be deleted from user_tenant table in mod-users.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/InitAPIs.java
[the remark this answers is not in the document]
Comment (SerhiiNosko) [reply src/main/java/org/folio/rest/impl/InitAPIs.java:1]: Basically we are implementing in the same manner as it was implemented in other modules like [This](https://github.com/folio-org/mod-source-record-manager/blob/master/mod-source-record-manager-server/src/main/java/org/folio/rest/impl/InitAPIImpl.java ) and [This](https://github.com/folio-org/mod-source-record-storage/blob/master/mod-source-record-storage-server/src/main/java/org/folio/rest/impl/InitAPIImpl.java)

I know that that implementation in these 2 modules works good, exists couple years, there were different performance tests for it. so decided to use the same pattern here

File: src/main/java/org/folio/verticle/consumers/ConsortiumDeleteEventsHandler.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  public ConsortiumDeleteEventsHandler(Vertx vertx) {
+    this.vertx = vertx;
+    this.userTenantService = new UserTenantService();
+  }
+
+  @Override
+  public Future<String> handle(KafkaConsumerRecord<String, String> record) {
+    Promise<String> result = Promise.promise();
+    List<KafkaHeader> kafkaHeaders = record.headers();
+    OkapiConnectionParams okapiConnectionParams = new OkapiConnectionParams(KafkaHeaderUtils.kafkaHeadersToMap(kafkaHeaders), vertx);
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/verticle/consumers/ConsortiumDeleteEventsHandler.java:35]: okapiConnectionParams now not used, so should be deleted from both consumer created/consumer deleted handlers

File: src/main/java/org/folio/repository/UserTenantRepository.java [hunk 2]
```diff
@@ -53,6 +54,13 @@ public Future<Boolean> saveUserTenant(Conn conn, UserTenant userTenant, String t
       .recover(throwable -> handleFailures(throwable, userTenant.getId()));
   }
 
+  public Future<Boolean> deleteUserTenant(Conn conn, UserTenant userTenant, String tenantId) {
+    String query = String.format(DELETE_SQL, convertToPsqlStandard(tenantId), USER_TENANT_TABLE_NAME);
+    Tuple queryParams = Tuple.of(userTenant.getUserId(), userTenant.getTenantId());
+    return conn.execute(query, queryParams).map(resultSet -> resultSet.size() == 1)
+      .recover(throwable -> handleFailures(throwable, userTenant.getId()));
```
[the remark this answers is not in the document]
Comment (SinghAdes) [reply src/main/java/org/folio/repository/UserTenantRepository.java]: fixed

File: src/main/java/org/folio/verticle/consumers/ConsortiumDeleteEventsHandler.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          event.getId(), event.getUserId(), event.getTenantId());
+        result.complete(event.getId());
+      })
+      .onFailure(e -> {
+        logger.error("Trying to delete of user primary affiliation event with consortium id: {} for user id: {} with tenant id: {} has been failed",
+          event.getId(), event.getUserId(), event.getTenantId(), e);
+        result.fail(e);
+      });
+
+    return result.future();
```
[the remark this answers is not in the document]
Comment (SinghAdes) [reply src/main/java/org/folio/verticle/consumers/ConsortiumDeleteEventsHandler.java]: fixed

**Review by SerhiiNosko — APPROVED** [review]

**Review by azizbekxm — APPROVED** [review]

**Review by tatsiana-tarhonskaya — APPROVED** [review]

**Pull request merged**